src/main/domain/vector-search.ts

```diff
@@ -0,0 +1,569 @@
+      await this.createTables();
+      
+      log.info("SQLite vector store initialized successfully");
+    } catch (error) {
+      log.error("Failed to initialize SQLite vector store:", error);
+      throw error;
+    }
+  }
+
+  private async loadSqliteVec() {
+    if (!this.db) return;
+
+    try {
+      // Try to load sqlite-vec extension
+      // Note: This requires the sqlite-vec extension to be available
+      // For now, we'll use basic SQLite with FTS5 for text search
+      log.info("sqlite-vec extension not available, using FTS5 for text search");
+    } catch (error) {
+      log.warn("Could not load sqlite-vec extension:", error);
+    }
+  }
+
+  private async createTables() {
+    if (!this.db) return;
+
+    return new Promise<void>((resolve, reject) => {
+      this.db!.serialize(() => {
+        // Create chunks table
+        this.db!.run(`
+          CREATE TABLE IF NOT EXISTS chunks (
+            id TEXT PRIMARY KEY,
+            recording_id TEXT NOT NULL,
+            chunk_index INTEGER NOT NULL,
+            text TEXT NOT NULL,
+            timestamp TEXT NOT NULL,
+            speaker TEXT NOT NULL,
+            start_time INTEGER NOT NULL,
+            end_time INTEGER NOT NULL,
+            embedding BLOB NOT NULL
+          )
+        `, (err) => {
+          if (err) {
+            log.error("Failed to create chunks table:", err);
+            reject(err);
+            return;
+          }
+        });
+
+        // Create FTS5 virtual table for text search
+        this.db!.run(`
+          CREATE VIRTUAL TABLE IF NOT EXISTS chunks_fts USING fts5(
+            text,
+            speaker,
+            timestamp,
+            content='chunks',
+            content_rowid='rowid'
+          )
+        `, (err) => {
+          if (err) {
+            log.error("Failed to create FTS5 table:", err);
+            reject(err);
+            return;
+          }
+        });
+
+        // Create triggers to keep FTS5 in sync
+        this.db!.run(`
+          CREATE TRIGGER IF NOT EXISTS chunks_ai AFTER INSERT ON chunks BEGIN
+            INSERT INTO chunks_fts(rowid, text, speaker, timestamp)
+            VALUES (new.rowid, new.text, new.speaker, new.timestamp);
+          END
+        `, (err) => {
+          if (err) {
+            log.error("Failed to create insert trigger:", err);
+            reject(err);
+            return;
+          }
+        });
+
+        this.db!.run(`
+          CREATE TRIGGER IF NOT EXISTS chunks_ad AFTER DELETE ON chunks BEGIN
+            INSERT INTO chunks_fts(chunks_fts, rowid, text, speaker, timestamp)
+            VALUES('delete', old.rowid, old.text, old.speaker, old.timestamp);
+          END
+        `, (err) => {
+          if (err) {
+            log.error("Failed to create delete trigger:", err);
+            reject(err);
+            return;
+          }
+          resolve();
+        });
+      });
+    });
+  }
+
+  async addTranscript(recordingId: string) {
```

Comment:
Maybe run `invalidateUiKeys` within this and other methods that change underlying data, to allow the UI to refetch what has changed.